### PR TITLE
 Compute and report maximum execution duration of currently running jobs.

### PR DESCRIPTION
**Pull request by mst-ableton**
ptal!

### CODE REVIEW

File: jenkins_exporter.py [hunk 1]
```diff
@@ -112,34 +123,28 @@ def _request_queue(self, jobs, config_mapping):
...
-            match_length = 0
-            for url, match in config_mapping.iteritems():
-                if task_url.startswith(url) and len(url) > match_length:
-                    job_name, config_name = match
-                    match_length = len(url)
-
-            if job_name is None or config_name is None:
+            job_name, job_config = map_url_to_job_config(task_url, config_mapping)
+
+            if job_name is None or job_config is None:
```
Comment (ala-ableton) [inline jenkins_exporter.py]: Renaming `job_name` to `job_config` could be done in a previous, distinct commit.

**Comment by mst-ableton** [comment]
@ala-ableton whoops, copy/paste error

**Comment by mst-ableton** [comment]
@ala-ableton renamed and split out the commits

**Comment by mst-ableton** [comment]
@ala-ableton rearranged!

File: jenkins_exporter.py [hunk 2]
```diff
@@ -54,39 +54,44 @@ def collect(self):
...
+        if result is None:
+            return [], {}
+        if DEBUG:
+            pprint(result)
 
-            jobs = []
-            for job in result['jobs']:
-                jobs.append(job)
-            return jobs
+        jobs = result['jobs']
```
Comment (ala-ableton) [inline jenkins_exporter.py]: Why is it fine to get `jobs` directly from the JSON now?

File: jenkins_exporter.py [hunk 3]
```diff
@@ -54,39 +54,44 @@ def collect(self):
...
 
-        jobs = parsejobs(url)
```
Comment (ala-ableton) [inline jenkins_exporter.py]: Why is this line deleted?

File: jenkins_exporter.py [hunk 4]
```diff
@@ -19,6 +19,17 @@
 NULL_CONFIGURATION_NAME = '__none__'
 
 
+def map_url_to_job_config(url, config_mapping):
+    """Return job and config name of the longest matching URL in the config mapping."""
```
Comment (ala-ableton) [inline jenkins_exporter.py]: Nitpicking, `Return job name and config`?

File: jenkins_exporter.py (hunk 2 above)
Comment (mst-ableton) [reply jenkins_exporter.py]: The [old code](https://github.com/lovoo/jenkins_exporter/commit/719db9cfcb0cc4941765614a0ad7d7bea4ce3875) handled "[folders](https://wiki.jenkins.io/display/JENKINS/CloudBees+Folders+Plugin)" which we don't use. We had to [remove the functionality](https://github.com/AbletonDevTools/jenkins_exporter/commit/c96070a1141bb2d8f82cdf9ee5d08fe0352ee96e) since it was breaking on our old Jenkins installation. At this point, the recursive parsejobs() isn't needed anymore, and it seemed like a good time to clean it up. I can split it out to a separate commit if you like.

File: jenkins_exporter.py (hunk 3 above)
Comment (mst-ableton) [reply jenkins_exporter.py]: See comment above

File: jenkins_exporter.py (hunk 2 above)
Comment (ala-ableton) [reply jenkins_exporter.py]: Please split it out and put that explanation in the commit message, thanks!

**Comment by mst-ableton** [comment]
@ala-ableton should be good to go again

**Comment by ala-ableton** [comment]
@mst-ableton GitHub doesn't show the commits in the right order... Anyway, LGTM :+1:

**Comment by mst-ableton** [comment]
🎉
